Titanium.Web.Proxy/Network/Tcp/TcpConnectionFactory.cs

```diff
@@ -71,88 +66,86 @@ internal class TcpConnectionFactory
+
+                            if (!string.IsNullOrEmpty(externalHttpsProxy.UserName) && externalHttpsProxy.Password != null)
+                            {
+                                await writer.WriteLineAsync("Proxy-Connection: keep-alive");
+                                await writer.WriteLineAsync("Proxy-Authorization" + ": Basic " + Convert.ToBase64String(Encoding.UTF8.GetBytes(externalHttpsProxy.UserName + ":" + externalHttpsProxy.Password)));
+                            }
+                            await writer.WriteLineAsync();
+                            await writer.FlushAsync();
+                            writer.Close();
                         }
-                        await writer.WriteLineAsync();
-                        await writer.FlushAsync();
-                        writer.Close();
-                    }
 
-                    using (var reader = new CustomBinaryReader(stream, server.BufferSize))
-                    {
-                        var result = await reader.ReadLineAsync();
-
-                        if (!new[] { "200 OK", "connection established" }.Any(s => result.ContainsIgnoreCase(s)))
+                        using (var reader = new CustomBinaryReader(stream, server.BufferSize))
                         {
-                            throw new Exception("Upstream proxy failed to create a secure tunnel");
-                        }
+                            var result = await reader.ReadLineAsync();
+
+                            if (!new[] { "200 OK", "connection established" }.Any(s => result.ContainsIgnoreCase(s)))
+                            {
+                                throw new Exception("Upstream proxy failed to create a secure tunnel");
+                            }
 
-                        await reader.ReadAndIgnoreAllLinesAsync();
+                            await reader.ReadAndIgnoreAllLinesAsync();
+                        }
+                    }
+                    else
+                    {
+                        client = new TcpClient(server.UpStreamEndPoint);
+                        await client.ConnectAsync(remoteHostName, remotePort);
+                        stream = new CustomBufferedStream(client.GetStream(), server.BufferSize);
                     }
-                }
-                else
-                {
-                    client = new TcpClient(server.UpStreamEndPoint);
-                    await client.ConnectAsync(remoteHostName, remotePort);
-                    stream = new CustomBufferedStream(client.GetStream(), server.BufferSize);
-                }
 
-                try
-                {
-                    sslStream = new SslStream(stream, true, server.ValidateServerCertificate,
-                        server.SelectClientCertificate);
+                    var sslStream = new SslStream(stream, false, server.ValidateServerCertificate, server.SelectClientCertificate);

```

Comment:
Was just wondering if we would have any benefit from setting this to false. Since we will be closing the inner-stream at the end of session and we call dispose() on this SslStream after that shouldn't that suffice?  Just so that for the sake of consistency with streamwriter/streamreader where we keep the innerstream open.